vehicle_gateway_integration_test/test/test_px4_multirobot.py

```diff
@@ -0,0 +1,232 @@
+                   '-z', z,
+                   '-R', roll,
+                   '-P', pitch,
+                   '-Y', yaw],
+        on_exit=[run_px4])
+
+    return [run_px4, spawn_entity]
+
+
+# This function specifies the processes to be run for our test
+@pytest.mark.launch_test
+def generate_test_description():
+    # This is necessary to get unbuffered output from the process under test
+    proc_env = os.environ.copy()
+    proc_env['PYTHONUNBUFFERED'] = '1'
+
+    world_pkgs = get_package_share_directory('vehicle_gateway_worlds')
+    gtw_models = get_package_share_directory('vehicle_gateway_models')
+
+    os.environ['GZ_SIM_RESOURCE_PATH'] = ':' + os.path.join(get_px4_dir(), 'models')
+    os.environ['GZ_SIM_RESOURCE_PATH'] += ':' + os.path.join(get_px4_dir(), 'worlds')
+    os.environ['GZ_SIM_RESOURCE_PATH'] += ':' + os.path.join(world_pkgs, 'worlds')
+    os.environ['GZ_SIM_RESOURCE_PATH'] += ':' + os.path.join(gtw_models, 'models')
+    os.environ['GZ_SIM_RESOURCE_PATH'] += ':' + os.path.join(gtw_models, 'configs_px4')
+
+    if 'SHOW_GZ_GUI' in os.environ and os.environ['SHOW_GZ_GUI']:
+        gz_gui_args = ''
+    else:
+        gz_gui_args = '--headless-rendering -s'
+    gz_args = f'{gz_gui_args} -r -v 4 empty_px4_world.sdf'
+
+    included_launch = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            [os.path.join(get_package_share_directory('ros_gz_sim'),
+                          'launch', 'gz_sim.launch.py')]),
+        launch_arguments=[('gz_args', [gz_args])]
+    )
+
+    autostart_magic_number = PythonExpression([
+        '{"x500": 4001, "rc_cessna": 4003, "standard_vtol": 4004}["',
+        LaunchConfiguration('vehicle_type'),
+        '"]'])
+
+    autostart_env_var = SetEnvironmentVariable(
+        'PX4_SYS_AUTOSTART',
+        autostart_magic_number)
+
+    micro_ros_agent = Node(
+        package='micro_ros_agent',
+        executable='micro_ros_agent',
+        arguments=['udp4', '-p', '8888'],
+        output='screen')
+
+    ld = launch.LaunchDescription([
+        DeclareLaunchArgument(
+            'vehicle_type',
+            default_value=['x500'],
+            description='Vehicle type',
+        ),
+        DeclareLaunchArgument(
+            'script_test',
+            default_value=[''],
+            description='Script to test',
+        ),
+        SetEnvironmentVariable('PX4_GZ_WORLD', ''),
+        SetEnvironmentVariable('PX4_GZ_MODEL', ''),
+        included_launch,
+        micro_ros_agent,
+        KeepAliveProc(),
+        # Tell launch to start the test
+        ReadyToTest(),
+        autostart_env_var
+    ])
+
+    [run_px4_1, spawn_1] = create_px4_instance('1')
+    [run_px4_2, spawn_2] = create_px4_instance('2')
+    ld.add_action(spawn_1)
+    ld.add_action(spawn_2)
+
+    context = {'included_launch': included_launch,
+               'micro_ros_agent': micro_ros_agent,
+               'run_px4_1': run_px4_1,
+               'run_px4_2': run_px4_2}
```

Comment:
Is there a better way to generalize this for `n` drones? Maybe build the context during iteration?